pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/channel/ServiceUnitStateChannelImpl.java

```diff
@@ -325,6 +325,16 @@ public synchronized void start() throws PulsarServerException {
                             ServiceUnitStateCompactionStrategy.class.getName()))
                     .create();
             tableview.listen((key, value) -> handle(key, value));
+            tableview.forEach((serviceUnit, data) -> {
+                if (debug) {
+                    log.info("Loaded the service unit state data. serviceUnit: {}, data: {}", serviceUnit, data);
+                }
+                ServiceUnitState state = state(data);
+                if (state.equals(Owned) && isTargetBroker(data.dstBroker())) {
+                    pulsar.getNamespaceService()
+                            .onNamespaceBundleOwned(LoadManagerShared.getNamespaceBundle(pulsar, serviceUnit));
+                }
+            });
```

Comment:
It feels like this should be using `tableview.forEachAndListen((key, value) -> handle(key, value));` instead?

Reply:
@heesung-sn @Demogorgon314 Please check this comment before merging.

Reply:
I dont think we should use foreachAndListen because we dont want the brokers to react on old messages. This can cause the unnecessary events in the channel.

Reply:
Yes, we should use the current way, and we may need to impl an internal table view for our case, there is a race condition between the table view create and listen, so it may miss some events.

Reply:
` there is a race condition between the table view create and listen, so it may miss some events`

I agree. Ideally, the table view constructor should accept the listeners to register them before receiving messages. I think forEachAndListen interface has the same problem too.

Reply:
> I agree. Ideally, the table view constructor should accept the listeners to register them before receiving messages. I think forEachAndListen interface has the same problem too.

+1

Reply:
> I agree. Ideally, the table view constructor should accept the listeners to register them before receiving messages. I think forEachAndListen interface has the same problem too.

How do the changes in this PR fix the issue?
It seems odd to have some code duplication in the `.forEach` handler and only handle a specific case instead of all possible cases. 

Why don't we fix the root cause in the table view?

Reply:
> @heesung-sn How do the changes in this PR fix the issue?

No. this PR does not fix the racing condition issue. I think we can track this as a separate issue, here https://github.com/apache/pulsar/issues/23158


> It seems odd to have some code duplication in the .forEach handler and only handle a specific case instead of all possible cases.

This is because the current table view's message handlers cannot differentiate "existing" and "tail" messages. The proposed solution in`.foreach` in this PR is partially duplicate logic to separately handle "existing" messages. I also mentioned this issue here, https://github.com/apache/pulsar/issues/23158

> Why don't we fix the root cause in the table view?

I assume this may require a separate PIP to improve the TableView interface. 


Currently, in the worst case, the race condition(`between the table view create and listen, so it may miss some events`) can be resolved by the leader's bundle state channel monitor, which should fix any invalid state eventually.

Reply:
I am not sure if this is a blocker for this PR, but I raised a PIP for this TableView update, https://github.com/apache/pulsar/pull/23167

Reply:
Should we merge this PR first and then after the TableView update is finished push another PR to fix the race condition issue?

Reply:
based on reading the current code of TableView, I don't yet understand why this PR isn't simply using `tableview.forEachAndListen((key, value) -> handle(key, value));`. I'll experiment to see if the tests would pass with that change.

Reply:
> I dont think we should use foreachAndListen because we dont want the brokers to react on old messages. This can cause the unnecessary events in the channel.

Let me give an example for this.

Currently, the topic has the following messages.
M1: bundle1, (Releasing from B1 to B2)
M2: bundle1, (Assigning to B2)
M3: bundle1, (Owned by B2)


When the topic does not run compaction yet, the table view can receive all non-compacted existing messages, M1-M3, when starting the table view on B1. Then, the broker B1 will unnecessarily close the Bundle, bundle1 if we pass 'handle()' to 'foreachAndListen'.

Reply:
@Demogorgon314 @heesung-sn In that case, I'd recommend refactoring the code so that `handle` is renamed to `handleEvent` and the function passed to the `.forEach` is moved to a new method `handleExisting` (or whatever naming describes this better).
Once the PIP for TableView improvements is processed, it can be migrated to use that. Makes sense?

Reply:
I agree with this change.

- handle() -> handleEvent()
- the lambda passed to foreach -> handleExisting()

And

> Once the PIP for TableView improvements is processed, it can be migrated to use that. Makes sense?

I agree.

Reply:
Updated.